src/logic/basic.lean

```diff
@@ -584,6 +584,14 @@ by simp [and_comm]
 @[simp] theorem exists_eq_right {a' : α} : (∃ a, p a ∧ a = a') ↔ p a' :=
 (exists_congr $ by exact λ a, and.comm).trans exists_eq_left
 
+@[simp] theorem exists_exists_and_eq_and {f : α → β} {p : α → Prop} {q : β → Prop} :
```

Comment:
I can imagine that there are 37 permutations of these lemmas, but maybe these two are the only ones that typically show up in a "simp-path"...

Reply:
These two come from `∃ y ∈ f '' s` and `∃ y ∈ set.range f`.